File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {

	public void solution() throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		
		Deque<Integer> queue = new ArrayDeque<Integer>();
		
		int N = Integer.valueOf(br.readLine());
		
		for (int i = 0; i < N; i++) {
			String[] command = br.readLine().split(" ");
			
			if(command.length == 2) {
				int number = Integer.valueOf(command[1]);
				queue.add(number);
			} else {
				switch (command[0]) {
				case "pop":
					if(queue.isEmpty()) {
						sb.append(-1 + "\n");						
					}else {						
						sb.append(queue.poll() + "\n");
					}
					break;
				case "size":
					sb.append(queue.size() + "\n");
					break;
				case "empty":
					if(queue.isEmpty()) {
						sb.append(1 + "\n");						
					}else {
						sb.append(0 + "\n");						
					}
					break;
				case "front":
					if(queue.isEmpty()) {
						sb.append(-1 + "\n");
					}else {
						sb.append(queue.peekFirst() + "\n");
					}
					break;
				case "back":
					if(queue.isEmpty()) {
						sb.append(-1 + "\n");
					}else {
						sb.append(queue.peekLast() + "\n");
					}
					break;
				}
			}
		}
		System.out.println(sb);
		
	}

	public static void main(String[] args) throws IOException {
		new Main().solution();
	}
}

File: 백준/Silver/10845. 큐/큐.js
const input = require('fs').readFileSync(process.version === 'v18.17.1' ? 'input.txt' : '/dev/stdin')
  .toString().trim().split("\n");

const solution = () => {
  const commands = input.slice(1);

  const queue = [];
  const result = [];

  for (const c of commands) {
    const [command, x] = c.split(" ");

    switch (command) {
      case "push":
        queue.push(x);
        break;
      case "pop":
        if(queue.length > 0) result.push(queue.shift());
        else result.push(-1);
        break;
      case "size":
        result.push(queue.length);
        break;
      case "empty":
        queue.length > 0 ? result.push(0) : result.push(1);
        break;
      case "front":
        if(queue.length > 0) result.push(queue.at(0));
        else result.push(-1);
        break;
      case "back":
        if(queue.length > 0) result.push(queue.at(-1));
        else result.push(-1);
        break;
      default:
        break;
    }
  }

  console.log(result.join('\n'));
};

solution();